File: Cajas3D/main.ts
function drawGift(size: number, symbol: string): string {
  const bg_size = size - 2;

  if (bg_size < 0) return "#\n";

  const top = " ".repeat(size - 1) + "#".repeat(size) + "\n";
  const center = "#".repeat(size) + symbol.repeat(bg_size) + "#" + "\n";
  const bottom = "#".repeat(size) + "\n";

  let top_content = "";
  let bottom_content = "";

  for (let i = 0; i < bg_size; i++) {
    const c =
      "#" + symbol.repeat(bg_size) + "#" + symbol.repeat(i) + "#" + "\n";

    top_content += " ".repeat(bg_size - i) + c;
    bottom_content += c.split("").reverse().join("");
  }

  bottom_content = bottom_content.split("").reverse().join("");

  const result = top + top_content + center + bottom_content + bottom;

  return result;
}

File: Cajas3D/main.py
def solution(size: int, symbol: str):
    bg_size = size - 2

    if bg_size < 0:
        return "#\n"
    
    top = " " * (size - 1) + "#" * size + "\n"
    center = "#" * size + symbol * bg_size + "#" + "\n"
    bottom = "#" * size + "\n"

    top_content = ""
    bottom_content = ""

    for i in range(bg_size):
        c = "#" + symbol * bg_size + "#" + symbol * i + "#" + "\n"
        top_content += " " * (bg_size - i) + c
        bottom_content += c[::-1]
    
    bottom_content = bottom_content[::-1]

    result = top + top_content + center + bottom_content + bottom

    return result


print(solution(20, "+"))